.github/actions/next-integration-stat/src/index.ts

```diff
@@ -136,35 +177,46 @@ function collectFailedTestResults(
       return true;
     })
     .map((logs) => {
-      let failedTest = logs.split(`failed to pass within`).shift();
-
-      // Look for the failed test file name
-      failedTest = failedTest?.includes("test/")
-        ? failedTest?.split("\n").pop()?.trim()
-        : "";
-
-      console.log("Failed test: ", { job: job.name, failedTest });
-
-      // Parse JSON-stringified test output between marker
-      try {
-        const testData = logs
-          ?.split("--test output start--")
-          .pop()
-          ?.split("--test output end--")
-          ?.shift()
-          ?.trim()!;
-
-        return {
-          job: job.name,
-          name: failedTest,
-          data: JSON.parse(testData),
-        };
-      } catch (_) {
-        console.log(`Failed to parse test data`);
-        return null;
+      let failedSplitLogs = logs.split(`failed to pass within`);
+      const ret = [];
+
+      while (!!failedSplitLogs && failedSplitLogs.length >= 1) {
```

Comment:
Can this ever be falsey?

Reply:
yes we consume it here https://github.com/vercel/turbo/pull/3659/files#diff-0e0c2f9cb52224d172853cfbed972f9ac25436bf53078850cd7b3a7168fef0e7R184 so eventually it'll drain